### PR TITLE
Core PBR Material

### PR DESCRIPTION
**Pull request by NickHardeman**
**Not Ready to be merged**. 
Failing check for VS, still needs fixing. Sending PR for visibility and help with VS and testing on other platforms.

Checks passing ( need to add back in some workflows, removed for testing).
- [x] msys2
- [x] emscripten
- [x] android
- [x] linux-64
- [x] linux-arm
- [x] vs
- [x] xcode
- [x] osx-make
- [x] iOS

Testing performed on:
- [x] OSX
- [ ] iOS
- [ ] Linux
- [ ] Raspberry PI
- [ ] Windows

Screen Shot 2022-11-22 at 1 25 34 AM

### CODE REVIEW

**Comment by ofTheo** [comment]
🎉🎉🎉🎉 
super exciting @NickHardeman !

File: libs/openFrameworks/gl/ofShader.cpp [hunk 1]
```diff
@@ -446,18 +447,10 @@ string ofShader::parseForIncludes( const string& source, vector<string>& include
...
-		}
-		else if (p != "pragma")
-			return false;
-
-		if (i.empty() || (f.size() < 2) ) 
-			return false;
+		if (p.empty() || i.empty() || (f.size() < 2) ) return false;
+		// -----| invariant: all tokens have values
 
+		if (p != "pragma") return false;
```
Comment (ofTheo) [inline libs/openFrameworks/gl/ofShader.cpp:453]: Just want to check, these changes don't undo @patriciogonzalezvivo's  lygia fixes?

Comment (NickHardeman) [reply libs/openFrameworks/gl/ofShader.cpp:453]: ah, that would be good to double check. I thought I used the nightly after his changes were merged.

File: libs/openFrameworks/gl/ofCubeMap.cpp [hunk 2]
```diff
@@ -0,0 +1,804 @@
...
+			data->bIrradianceAllocated = false;
+			glDeleteTextures(1, &data->irradianceMapId );
+		}
+		if( data->bCubeMapAllocated ) {
+			data->bCubeMapAllocated = false;
+			glDeleteTextures(1, &data->cubeMapId );
+		}
+	}
+}
+
```
Comment (ofTheo) [inline libs/openFrameworks/gl/ofCubeMap.cpp:118]: I wonder if we need a copy constructor for this? 
And/or texture reference counting ( retain ) similar to: 

https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/gl/ofTexture.cpp#L187-L248

Comment (NickHardeman) [reply libs/openFrameworks/gl/ofCubeMap.cpp:118]: Yeah would probably be beneficial

**Comment by ofTheo** [comment]
Everything else looks great! 
Thanks @NickHardeman!

**Comment by ofTheo** [comment]
🎉👏👏👏

**Pull request merged**